crates/bevy_gltf/src/loader.rs

```diff
@@ -1653,25 +1662,19 @@ async fn load_buffers(
 }
 
 fn resolve_node_hierarchy(
-    nodes_intermediate: Vec<(GltfNode, Vec<usize>)>,
-    asset_path: &Path,
-) -> Vec<GltfNode> {
-    let mut has_errored = false;
+    nodes_intermediate: Vec<(GltfNode, Vec<(usize, Handle<GltfNode>)>)>,
+) -> Result<Vec<GltfNode>, GltfError> {
     let mut empty_children = VecDeque::new();
     let mut parents = vec![None; nodes_intermediate.len()];
     let mut unprocessed_nodes = nodes_intermediate
         .into_iter()
         .enumerate()
         .map(|(i, (node, children))| {
-            for child in &children {
```

Comment:
Much of defensive programming here is not needed because gLTF file won't load if the children are invalid.